cli/src/main/java/keywhiz/cli/commands/AddAction.java

```diff
@@ -139,6 +139,7 @@ private void createAndAssignSecret(String secretName, String description, byte[]
 
   private byte[] readSecretContent() {
     try {
+      System.out.print("Enter secret content, Ctrl+D when done: ");
       byte[] content = ByteStreams.toByteArray(stream);
       if (content.length == 0) {
         throw new RuntimeException("Secret content empty!");
```

Comment:
What effect does this have if you pipe the secret in? Is there a way we can only emit this if we're actually waiting for user input?

Reply:
Great question, Michael. I tested with following CLI about piping: `echo "xxx" | java -jar "$KEYWHIZ_CLI_JAR" -U "$KEYWHIZ_SERVER_URL" "$@" --devTrustStore add --name secretName secret`. The behavior is that the CLI will finish with success. A better implementation will check whether stream is waiting for user input or data is already available. However, there seems no reliable method to check for the status: https://stackoverflow.com/questions/5826070/java-runtime-process-check-if-waiting-for-input